src/Features/Core/Portable/AddImport/AbstractAddImportFeatureService.cs

```diff
@@ -376,10 +379,22 @@ private static bool IsHostOrRemoteWorkspace(Project project)
         /// </summary>
         private static bool IsInPackagesDirectory(PortableExecutableReference reference)
         {
-            return ContainsPathComponent(reference, "packages")
-                || ContainsPathComponent(reference, "packs")
-                || ContainsPathComponent(reference, "NuGetFallbackFolder")
-                || ContainsPathComponent(reference, "NuGetPackages");
+            var metadataId = reference.GetMetadataId();
+            if (!s_isInPackagesDirectory.TryGetValue(metadataId, out var result))
```

Comment:
>s_isInPackagesDirectory [](http://example.com/codeflow?start=17&length=23)

I assume we don't need to worry about cache size or invalidation

Reply:
Correct.  Even if there were a thousand unique references, it would be a dictionary with 1000 items in it (so nothing).  :)